packages/jest-changed-files/src/hg.ts

```diff
@@ -20,7 +20,7 @@ const adapter: SCMAdapter = {
 
     const args = ['status', '-amnu'];
     if (options && options.withAncestor) {
-      args.push('--rev', `min((!public() & ::.)+.)^`);
+      args.push('--rev', 'min((not public() and ancestors(.))^ or .^)');
```

Comment:
The `^` is left in place, but not at the end. Also I replaced `!`, `&`, `+` with `not`, `and`, `or`. Might be these are less problematic.

Reply:
@mjesun @quark-zju dunno if any of you are still at FB (I know Miguel is not, at least 🙂), but does this change make sense to you? Could you verify it?

Less weird syntax, but I've never written a mercurial command in my life, so I have no frame of reference to know if this is the correct fix or not 😅

Reply:
It might be this is not the same thing with `^` moved inside `min()`. But leaving `^` at the end of command throws `abort: empty revision range` locally and in CI. Strange.